src/connectionimpl.cpp

```diff
@@ -123,7 +123,9 @@ uint64_t ConnectionImpl::parse(const Buffer &buffer)
         try
         {
             // try to recognize the frame
-            ReceivedFrame receivedFrame(ReducedBuffer(buffer, processed), _maxFrame);
```

Comment:
Why do you think that It is an undefined behaviour?

12.2/5 "Temporary objects":

> The temporary to which the reference is bound or the temporary that is the complete object to a subobject of which the temporary is bound persists for the lifetime of the reference except as specified below. A temporary bound to a reference member in a constructor’s ctor-initializer (12.6.2) persists until the constructor exits. **A temporary bound to a reference parameter in a function call (5.2.2) persists until the completion of the full expression containing the call.**